### PR TITLE
RUBY-1111 Support custom GridFS file IDs

### PR DESCRIPTION
**Pull request by estolfo**

### CODE REVIEW

**Comment by estolfo** [comment]
@rstam I'd welcome a code review, you have the chance. Thanks = )

File: lib/mongo/grid/fs_bucket.rb [hunk 1]
```diff
@@ -366,12 +367,13 @@ def open_upload_stream(filename, opts = {})
...
       #
       # @example Upload a file to the GridFS bucket.
-      #   fs.upload_from_stream('a-file.txt')
+      #   fs.upload_from_stream('a-file.txt', file)
       #
       # @param [ String ] filename The filename of the file to upload.
       # @param [ IO ] io The source io stream to upload from.
       # @param [ Hash ] opts The options for the write stream.
       #
+      # @option opts [ Integer ] :file_id An optional unique file id. An ObjectId is generated otherwise.
```
Comment (rstam) [inline lib/mongo/grid/fs_bucket.rb]: Type should be Object

File: lib/mongo/grid/stream/write.rb [hunk 2]
```diff
@@ -27,7 +27,7 @@ class Write
           # @since 2.1.0
           attr_reader :fs
 
-          # @return [ BSON::ObjectId ] file_id The id of the file being uploaded.
+          # @return [ Obejct ] file_id The id of the file being uploaded.
```
Comment (rstam) [inline lib/mongo/grid/stream/write.rb]: Typo. Should be "Object"

**Comment by estolfo** [comment]
thanks @rstam !

**Pull request merged**